### PR TITLE
Widgets: Remove deprecations and make arguments keyword only

### PR DESCRIPTION
**Pull request by oscargus**
## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
@oscargus do you mind holding off on any other deprecation removals? We would like to leave that as a good first issue for the sprint tomorrow.

**Comment by oscargus** [comment]
Sure, no problems! Didn't know that.

**Comment by QuLogic** [comment]
No worries, we only decided yesterday.

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -1283,60 +1282,6 @@ def disconnect(self, cid):
         """Remove the observer with connection id *cid*."""
         self._observers.disconnect(cid)
 
-    @_api.deprecated("3.7",
-                     addendum="Any custom property styling may be lost.")
-    @property
-    def rectangles(self):
-        if not hasattr(self, "_rectangles"):
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:1290]: There's still a `hasattr` for this attribute in `_clicked.

File: lib/matplotlib/widgets.py [hunk 2]
```diff
@@ -1283,60 +1282,6 @@ def disconnect(self, cid):
...
-        if not hasattr(self, "_lines"):
-            with _api.suppress_matplotlib_deprecation_warning():
-                _ = self.lines
-        return self._rectangles
-
-    @_api.deprecated("3.7",
-                     addendum="Any custom property styling may be lost.")
-    @property
-    def lines(self):
-        if not hasattr(self, "_lines"):
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:1314]: There's still a `hasattr` check in `_clear` and `set_active`.

File: lib/matplotlib/widgets.py [hunk 3]
```diff
@@ -1856,23 +1799,6 @@ def disconnect(self, cid):
         """Remove the observer with connection id *cid*."""
         self._observers.disconnect(cid)
 
-    @_api.deprecated("3.7",
-                     addendum="Any custom property styling may be lost.")
-    @property
-    def circles(self):
-        if not hasattr(self, "_circles"):
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:1863]: The `hasattr` is still checked in `_clear`, `_clicked`, and `set_active`.

File: lib/matplotlib/widgets.py [hunk 4]
```diff
@@ -1742,16 +1661,10 @@ def _clicked(self, event):
...
-                        or np.linalg.norm(pclicked - p.center) < p.radius):
-                    distances[i] = np.linalg.norm(pclicked - p.center)
-        else:
-            for i, t in enumerate(self.labels):
-                if (i in inds["ind"]
-                        or t.get_window_extent().contains(event.x, event.y)):
-                    distances[i] = np.linalg.norm(pclicked - coords[i])
+        for i, t in enumerate(self.labels):
+            if (i in inds["ind"]
+                    or t.get_window_extent().contains(event.x, event.y)):
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:1666]: Fits in a line now:
```suggestion
            if i in inds["ind"] or t.get_window_extent().contains(event.x, event.y):
```

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/widgets.py (hunk 4 above)
Comment (anntzer) [reply lib/matplotlib/widgets.py:1666]: I have a further PR that rewrites this loop, so let's just get this in for now.

**Review by anntzer — APPROVED** [review]

**Pull request merged**